### ISSUE TITLE
implement PEP 554 for use in test suite

### ISSUE THREAD

**Issue by ericsnowcurrently** [ericsnowcurrently/multi-core-python#21]
Issue #22 is about making use of subinterpreters (specifically the PEP 554 API) in the CPython test suite.
Until PEP 554 is accepted (or if it isn't) we should add a PEP 554 implementation as `Lib/test/support/_interpreters.py` (e.g. `test.support._interpreters`).

**Comment by ericsnowcurrently** [comment]
@nanjekyejoannah, what's the status on this?  IIRC you were blocked on a few other issues.  I'm hoping we can get them resolved in the next week or so and then get this merged.  What do you think?

**Comment by nanjekyejoannah** [comment]
@ericsnowcurrently, yes I have most of it working. I will need to take a day or two for final touches. I have exams this coming week till around 24th but I will be able to get the remaining work through after 24th. So we are talking about practically hearing from me by 28th.

**Comment by ericsnowcurrently** [comment]
Sounds good.  Good luck on your exams! 😁

**Comment by nanjekyejoannah** [comment]
Blocked by  [this PR](https://github.com/python/cpython/pull/17323). I think @LewisGaul has looked at this longer.  It makes sense for you to finish. @LewisGaul  happy to know where to help.

**Comment by nanjekyejoannah** [comment]
@ericsnowcurrently  I opened a PR to solve this here : https://github.com/python/cpython/pull/19985

**Comment by ericsnowcurrently** [comment]
Awesome!  I should have some time to look tomorrow.
